packages/web-ui/components/create-room-form.tsx

```diff
@@ -0,0 +1,109 @@
+'use client';
+
+import { useState } from 'react';
+import { zodResolver } from '@hookform/resolvers/zod';
+import { Lock, Plus } from 'lucide-react';
+import { useForm } from 'react-hook-form';
+import * as z from 'zod';
+
+import { CreateRoomRequestSchema } from '@/types/rooms/create-room.request.dto';
+import { CreateRoomRequestDtoType } from '@/types/rooms/create-room.request.dto.d';
+import { Button } from '@/components/ui/button';
+import { Checkbox } from '@/components/ui/checkbox';
+import {
+  Dialog,
+  DialogContent,
+  DialogDescription,
+  DialogHeader,
+  DialogTitle,
+  DialogTrigger,
+} from '@/components/ui/dialog';
+import {
+  Form,
+  FormControl,
+  FormField,
+  FormItem,
+  FormLabel,
+  FormMessage,
+} from '@/components/ui/form';
+import { Input } from '@/components/ui/input';
+
+const formSchema = z.object({
```

Comment:
I think you can pick up the schema from the common types and extend it with your stuff. At least with Yup it's possible